data-prepper-plugins/mutate-event-processors/src/test/java/org/opensearch/dataprepper/plugins/processor/mutateevent/AddEntryProcessorTests.java

```diff
@@ -355,13 +356,41 @@ public void testKeyIsNotAdded_when_addWhen_condition_is_false() {
         assertThat(editedRecords.get(0).getData().containsKey("newMessage"), is(false));
     }
 
+    @Test
+    public void testMetadataKeySet() {
```

Comment:
This test name should describe the test. Please be more explicit. 

Please add other tests which test `overwrite_if_key_exists`, `format`, `value` and any failure scenario that comes with using these options with the `metadata_key`. I know we are leveraging the some of the same code under the hood but our unit tests should harden the contract that `metadata_key` works with these other configuration options. This will help prevent future developers breaking support for these features unknowingly and bugs being released. These unit tests will catch any breaking changes.

Reply:
@cmanning09  I am thinking if we should do something like this for configuration to avoid duplicate config options
```
processor:
    - add_entries:
        entries:
        - key: "newMessage"
           format: "new ${message}"
           overwrite_if_key_exists: true
           add_to_metadata: true
```
This will keep the existing functionality/options as is but adds a new boolean flag that will add to metadata instead of the event. I can't think of a case where same thing needs to be added to both event and metadata